Sources/Core/ConditionalCompilationFactors.swift

```diff
@@ -0,0 +1,35 @@
+import Utils
+
+/// The factors that influence conditional compilation.
+public struct ConditionalCompilationFactors: Codable {
+
+  /// The target operating system.
+  let operatingSystem: Host.OperatingSystem
+
+  /// The target architecture.
+  let architecture: Host.Architecture
+
+  /// The version of the compiler.
+  let compilerVersion: SemanticVersion
+
+  /// The version of the Hylo language version we are targeting.
+  let hyloVersion: SemanticVersion
+
+  /// `true` if we are compiling for freestanding mode.
```

Comment:
Try not to write a comment that requires the reader to already understand the meaning of the name you chose.

```suggestion
  /// `true` if the standard library exposes only non-OS-dependent parts.
```